client/state/analytics/actions/with-analytics.js

```diff
@@ -14,6 +9,18 @@ const joinAnalytics = ( analytics, action ) =>
 				dispatch( analytics );
 				dispatch( action );
 		  }
-		: merge( {}, action, { meta: { analytics: mergedMetaData( analytics, action ) } } );
+		: {
+				...action,
+				...{
+					meta: {
+						...action?.meta,
+						analytics: mergedMetaData( analytics, action ),
+					},
+				},
+		  };
 
-export const withAnalytics = curry( joinAnalytics );
+export const withAnalytics = ( ...args ) => {
+	return args.length >= joinAnalytics.length
+		? joinAnalytics( ...args )
+		: ( ...args2 ) => withAnalytics( ...args.concat( args2 ) );
+};
```

Comment:
This is easier to read IMO:
```js
function withAnalytics( analytics, action ) {
  if ( action === undefined ) {
    return ( a ) => joinAnalytics( analytics, a );
  }
  return joinAnalytics( analytics, action );
}
```

How hard would it be to choose just one convention for `withAnalytics` (either curried or not) and codemod all usages?

Typing the curried version will be a bit of a hassle.

Reply:
> This is easier to read IMO:

Agreed, thanks! Updated in https://github.com/Automattic/wp-calypso/pull/51895/commits/465c252d5c22b8dcc677700177f026fdc2e6f562

> How hard would it be to choose just one convention for withAnalytics (either curried or not) and codemod all usages?

I haven't considered that, but I think it shouldn't be a huge deal. Sounds like work for another PR though 😉